server/expression.js

```diff
@@ -167,6 +182,51 @@ expression.getTfaProfile_ = function(fileName, geneNames, conditionNames) {
 };
 
 /**
+ * Reads the TFA matrix in text file.
+ * @param {string} fileName Path to the TFA matrix file.
+ * @return {!expression.Matrix}
+ * @private
+ */
+expression.readTfaMatrix_ = function(fileName) {
```

Comment:
Can we change readMatrix_ so that it accepts all rows when geneNames is null, and accepts all columns when conditionNames is null? Otherwise we have too many dup code.